backend/infrahub/core/diff/query_parser.py

```diff
@@ -551,6 +554,14 @@ def _apply_relationship_previous_values(
                     DatabaseEdgeType(p.property_type): None for p in property_set
                 }
                 base_diff_property_by_type[DatabaseEdgeType.IS_RELATED] = None
+                latest_diff_property_times_by_type: dict[DatabaseEdgeType, Timestamp] = {}
+                for diff_property in property_set:
+                    if (
+                        diff_property.property_type not in latest_diff_property_times_by_type
+                        or diff_property.changed_at > latest_diff_property_times_by_type[diff_property.property_type]
+                    ):
+                        latest_diff_property_times_by_type[diff_property.property_type] = diff_property.changed_at
+
```

Comment:
this new logic updates how we apply the changes on the base branch to the diff branch when setting previous values in the diff. the problem was that if the changes on `main` occurred after the changes on `branch`, then we could end up treating later changes as though they preceded the changes on the branch.
this new dict and conditional to check on it fix that by ensuring that changes on main that occurred after the changes on the branch are ignored